### ISSUE TITLE
various charts don't fit with auto-height and unsized div

### ISSUE THREAD

**Issue by gordonwoodhull** [dc-js/dc.js#1070]
When using the feature to automatically determine the chart size (by not setting `width`/`height` or setting them to null), and the containing div does not have an absolute size, the size of the pie can be calculated wrong and it gets clipped:

![image](https://cloud.githubusercontent.com/assets/1366709/11932846/3911605a-a7c6-11e5-9f1b-a9060f25344a.png)

```html
<div id="test"></div>
```

```{js}
var chart = dc.pieChart("#test");
  chart
    .slicesCap(4)
    .innerRadius(50)
    .dimension(runDimension)
    .group(speedSumGroup)
    .legend(dc.legend());
```

This is because the SVG size is based on the div, then the div sizes to the SVG, and then when `.height()` gets read again in order to set the radius, now the auto-height is a little bigger.

**Comment by gordonwoodhull** [comment]
Even worse if there's other stuff in the div (like controls)

![image](https://cloud.githubusercontent.com/assets/1366709/11933134/e926c326-a7c7-11e5-8f3b-b823a19da736.png)

**Comment by gordonwoodhull** [comment]
Similarly with the row chart:

![image](https://cloud.githubusercontent.com/assets/1366709/11933321/33cc2866-a7c9-11e5-9a65-a3b0a037a241.png)

```html
<div id="chart-row-spenders">
<p>Some Controls.</p></div>
```

```js
var spenderRowChart = dc.rowChart("#chart-row-spenders");
spenderRowChart
    .dimension(nameDim)
    .group(spendPerName)
    .elasticX(true);
```

**Comment by gordonwoodhull** [comment]
heatmap (data goes to 1985 and x axis should be displayed)

**Comment by gordonwoodhull** [comment]
Solution: only calculate the size once per render. Fixed in 2.0 beta 24.
